cmd/main.go

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/m3db/prometheus_remote_client_golang/promremote"
+)
+
+type tagList []promremote.Tag
+type dp promremote.Datapoint
+
+func main() {
+	var (
+		writeURLFlag string
+		tagsListFlag tagList
+		dpFlag       dp
+	)
+
+	flag.StringVar(&writeURLFlag, "u", promremote.DefaultRemoteWrite, "remote write endpoint")
+	flag.Var(&tagsListFlag, "t", "tag pair to include in metric. specify as key:value e.g. status_code:200")
+	flag.Var(&dpFlag, "d", "datapoint to add. specify as value(float),unixTimestamp(int) e.g. 14.23,1556026059. use `now` instead of timestamp for current time")
+
+	flag.Parse()
+
+	tsList := promremote.TSList{
+		{
+			Tags:      []promremote.Tag(tagsListFlag),
+			Datapoint: promremote.Datapoint(dpFlag),
+		},
+	}
+
+	cfg := promremote.NewConfig(
+		promremote.WriteURLOption(writeURLFlag),
+	)
+
+	client, err := promremote.NewClient(cfg)
+	if err != nil {
+		log.Fatal(fmt.Errorf("unable to construct client: %v", err))
+	}
+
+	if err := client.WriteTimeSeries(context.Background(), tsList); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func (t *tagList) String() string {
+	return ""
```

Comment:
Not entirely sure what to put here - doesn't really make sense to put anything (unless I'm not understanding something about this interface I need to satisfy).

Reply:
eh, if you're going to define the method, it should do what it's supposed to do (i know, annoying in this case): return a string representation of the `tagList`.

Reply:
okay let me work on it and see what makes sense.

Reply:
Easiest way:
```go
func (t *tagList) String() {
  var tags [][]string
  for _, v := range t {
    tags = append(tags, []string{v.Name, v.Value})
  }
  return fmt.Sprintf("%v", tags)
}
```

Example:
https://play.golang.org/p/I6qkCg1XfdX

Output:
```
[[foo bar] [baz qux]]
```

Reply:
or

```go
func (t tagList) String() {
  tags := make([]string, len(t))
  for i, tag := range t {
    tags[i] = fmt.Sprintf("%s=%s", tag.Name, tag.Value)
  }

  return strings.Join(tags, ",")
}
```

output:

    foo=bar,baz=bat

whichever you like, doesn't matter a ton.